tutorials/metrics.md

```diff
@@ -39,22 +39,34 @@ metrics_key = "/path/to/key.pem"
+| `crio_containers_oom`                            | `name`                                                                                                                                  | Counter   | (DEPRECATED) Containers killed because they ran out of memory (OOM) by their name                                 |
+| `crio_processes_defunct`                         |                                                                                                                                         | Gauge     | Total number of defunct processes in the node                                                                     |
+| `crio_operations_total`                          | every CRI-O RPC\*                                                                                                                       | Counter   | Cumulative number of CRI-O operations by operation type.                                                          |
+| `crio_operations_latency_seconds_total`          | every CRI-O RPC\*,<br><br>`network_setup_pod` (CNI pod network setup time),<br><br>`network_setup_overall` (Overall network setup time) | Summary   | Latency in seconds of CRI-O operations. Split-up by operation type.                                               |
+| `crio_operations_latency_seconds`                | every CRI-O RPC\*                                                                                                                       | Gauge     | Latency in microseconds of individual CRI calls for CRI-O operations. Broken down by operation type.              |
+| `crio_operations_errors_total`                   | every CRI-O RPC\*                                                                                                                       | Counter   | Cumulative number of CRI-O operation errors by operation type.                                                    |
+| `crio_image_pulls_by_digest_bytes_total`         | `name`, `digest`, `mediatype`, `size`                                                                                                   | Counter   | Bytes transferred by CRI-O image pulls by digest.                                                                 |
+| `crio_image_pulls_by_name_bytes_total`           | `name`, `size`                                                                                                                          | Counter   | Bytes transferred by CRI-O image pulls by name.                                                                   |
+| `crio_image_pulls_by_name_skipped_bytes_total`   | `name`                                                                                                                                  | Counter   | Bytes skipped by CRI-O image pulls by name.                                                                       |
```

Comment:
Not sure what this means, might make sense to improve name and/or description. Is this due to image reuse? Are skipped bytes included in bytes_total? Is `crio_image_pulls_by_name_skipped_bytes_total/crio_image_pulls_by_name_bytes_total` somewhat like a "cache hit" ratio?

Reply:
yeah that's a good way of thinking about it

Reply:
@haircommander @discordianfish So we can keep both of these metrics with `_bytes_total` suffix and remove the `_by_name` part as well, right?

Reply:
Also, am now worried about the cardinality of `name` label, should that be much concern?

Reply:
yeah we'll have to make a hard decision about the crio_image ones. If we do any reporting on the image level we're at risk of high cardinality. I don't *think* this number will get huge (order of 100s?), but I am not personally aware of what constitutes high

Reply:
Yeah not sure about the name either. On the most system this should be max in the 100s per node, but nothing might prevent users from pulling millions of images. Beside being a problem to ingest to many metrics, crio will also hold these in memory until it gets restarted.

Reply:
Oh and not sure if that was clear: Skipped bytes should be included in bytes_total if not and the doc should clarify that. Not sure if that's the case already but it makes calculating ratios more convenient.

Reply:
> Oh and not sure if that was clear: Skipped bytes should be included in bytes_total if not and the doc should clarify that. Not sure if that's the case already but it makes calculating ratios more convenient.

@discordianfish you mean exposing a metric like `skip_ratio` be more convenient that `skipped_bytes`, right?

Reply:
@swghosh No, I'm just saying that skipped bytes should be included in total bytes so that `crio_image_pulls_by_name_skipped_bytes_total/crio_image_pulls_by_name_bytes_total` would calculate the ratio. Might be already the case in which I'd just point that out in the docs here.

Reply:
https://github.com/cri-o/cri-o/blob/854e2fda2f261efcce3744888724d97f348602be/server/image_pull.go#L148-L220
I think total pulled bytes also includes skipped bytes.

Reply:
We can include about the skip ratio in doc. (that: `image_pulls_skipped_bytes_total / image_pulls_bytes_total` is actually a ratio)

Reply:
Included in 8ec8b7396.
https://github.com/cri-o/cri-o/blob/8ec8b7396e018ad08f28b159a235aaff06d6287e/tutorials/metrics.md#L49